lib/ferrum/target.rb

```diff
@@ -59,9 +67,27 @@ def window?
       !!opener_id
     end
 
+    def page?
+      @params["type"] == "page"
+    end
+
+    def worker?
+      @params["type"] == "worker"
+    end
+
     def maybe_sleep_if_new_window
       # Dirty hack because new window doesn't have events at all
       sleep(NEW_WINDOW_WAIT) if window?
     end
+
+    def connection(**options)
```

Comment:
Not sure what the proper naming would be for this if it can be both a page and a worker.